docs/middleware/README.md

```diff
@@ -0,0 +1,64 @@
+- Registration/Deregistration conditions
+
+These two points have been addressed through the Registry Coordinator/Registry architecture.
+
+### Quorums
+
+Before we dive in, we need to define what a quorum is. Quorums are the different divisions of the operator set for an AVS. One can think of a quorum simply being defined by the token staked for that quorum, although [it slightly more complicated than that](./StakeRegistry.md#definitions). One often wants to make trust assumptions on quorums, but wants many quorums for the same AVS. 
+
+One example of the quorum concept is in EigenDA, where we have a single ETH quorum for which LSTs and native beacon chain ETH are accepted as stake and another quorum for each rollup that wants to stake their own token for security.
+
+### RegistryCoordinator
+The Registry Coordinator is a contract that is deployed by each AVS. It handles
+- Keeping track of what quorums operators are a part of
+- Handling operator churn (registration/deregistration)
+- Communicating to registries
+The current implementation of this contract is the [BLSRegistryCoordinatorWithIndices](./BLSRegistryCoordinatorWithIndices.md).
+
+
+### Registries
+The registries are contracts that keep track of the attributes of individual operators. For example, we have initially built the 
+- [StakeRegistry](./StakeRegistry.md) which keeps track of the stakes of different operators for different quorums at different times
+- [BLSPubkeyRegistry](./BLSPubkeyRegistry.md) which keeps track of the aggregate public key of different quorums at different times
+- [IndexRegistry](./IndexRegistry.md) which keeps track of an ordered list of the operators in each quorum at different times. (note that this behavior is likely only needed for EigenDA)
+Registries are meant to be read from and indexed by offchain AVS actors (operators and AVS coordinators). 
+
+A registry coordinator has 1 or more registries connected to it and all of them are called when an operator registers or deregisters. They are a plug and play system that are meant to be added to the RegistryCoordinator as needed. AVSs should create registries they need for their purpose.
+
+## TODO: Service Manager
+
+## BLS Signature Checker
+
+At the core of many AVSs on EigenLayer (almost all except those that effect Ethereum block production) is the verification of a quorum signature of an AVS's operator set on a certain message and slashing if some quality of that message and other state is true. The registry architechture is optimized for making this signature as cheap as possible to verify (it is still relatively expensive).
+
+The current implementation of this contract is the [BLSSignatureChecker](./BLSSignatureChecker.md).
+
+## Further reading
+
+More detailed functional docs have been written on the AVS architechture implemented in the middleware contracts. The recommended order for reading the other docs in this folder is 
```

Comment:
typo, misspelling of 'architecture' as 'architechture'

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/153/commits/835792608127dc41815d62b928313724dfb287b7